include/aikido/control/Executor.hpp

```diff
@@ -0,0 +1,148 @@
+#ifndef AIKIDO_CONTROL_EXECUTOR_HPP_
+#define AIKIDO_CONTROL_EXECUTOR_HPP_
+
+#include <chrono>
+#include <future>
+#include <set>
+#include <vector>
+
+#include <dart/dynamics/DegreeOfFreedom.hpp>
+#include <dart/dynamics/Skeleton.hpp>
+
+#include "aikido/common/ExecutorThread.hpp"
+#include "aikido/common/pointers.hpp"
+
+namespace aikido {
+namespace control {
+
+AIKIDO_DECLARE_POINTERS(Executor)
+
+enum class ExecutorType
+{
+  kSTATE = -1,
```

Comment:
Since this is already an enum, I think adding the `k` prefix to all these names is redundant.